### PR TITLE
Update file license header from Apache2 to MIT

### PR DESCRIPTION
**Pull request by JoeRobich**
Resolves https://github.com/dotnet/roslyn/issues/50604

### CODE REVIEW

File: eng/InternalTools.props [hunk 1]
```diff
@@ -1,6 +1,6 @@
-<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
 <Project>
-   
+
```
Comment (dibarbet) [inline eng/InternalTools.props:3]:
```suggestion
```

**Review by dibarbet — APPROVED** [review]

File: eng/InternalTools.props (hunk 1 above)
Comment (JoeRobich) [reply eng/InternalTools.props:3]: Sigh, I was trying so hard not to mess with trailing spaces that I introduced one. Now to find out why VSCode wasn't telling me that happened.

**Review by sharwell — APPROVED** [review]

**Review by jmarolf — APPROVED** [review]

**Review by 333fred — APPROVED** [review]

File: src/Compilers/VisualBasic/Portable/Symbols/EmbeddedSymbols/VbMyTemplateText.vb [hunk 2]
```diff
@@ -1,4 +1,6 @@
-﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
```
Comment (Youssef1313) [inline src/Compilers/VisualBasic/Portable/Symbols/EmbeddedSymbols/VbMyTemplateText.vb:3]: How IDE0073 wasn't generated for mismatching header template?
I see it's set as a warning in `.editorconfig`.

**Comment by Youssef1313** [comment]
Does this also fix #25423?

File: eng/common/internal/Tools.csproj [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
-<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
+<!-- Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE file in the project root for more information. -->
```
Comment (Youssef1313) [inline eng/common/internal/Tools.csproj:2]: The readme in eng/common says to not modify files here.

https://github.com/dotnet/roslyn/blob/e959d2a9c9b27cde09430f7755ff48bc2825d66e/eng/common/README.md

Comment (JoeRobich) [reply eng/common/internal/Tools.csproj:2]: Good catch! I will revert.

File: src/Compilers/VisualBasic/Portable/Symbols/EmbeddedSymbols/VbMyTemplateText.vb (hunk 2 above)
Comment (Youssef1313) [inline src/Compilers/VisualBasic/Portable/Symbols/EmbeddedSymbols/VbMyTemplateText.vb:3]: The current main already has the correct header here. Can you please rebase on the current main?

File: src/Compilers/VisualBasic/Test/Syntax/Resources/AllInOne.vb [hunk 4]
```diff
@@ -1,4 +1,6 @@
-﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+﻿' Licensed to the .NET Foundation under one or more agreements.
```
Comment (Youssef1313) [inline src/Compilers/VisualBasic/Test/Syntax/Resources/AllInOne.vb:1]: This is also already correct in the current main.

**Pull request merged**